### PR TITLE
Numeric Stats Mixin: `pop` disabled elements from profile report 

### PR DESCRIPTION
**Pull request by taylorfturner**
- pop disabled elements from profile
- add `test_report` for testing new functionality

### CODE REVIEW

File: dataprofiler/profilers/numerical_column_stats.py [hunk 1]
```diff
@@ -292,6 +292,37 @@ def profile(self):
...
+        profile = self.profile()
+
+        if remove_disabled_flag:
+            profile_keys = list(profile.keys())
+            for profile_key in profile_keys:
+                if profile_key in ['mode', 'quantiles', 'histogram']:
+                    if 'histogram_and_quantiles' in calcs_dict_keys:
+                        continue
+                elif profile_key == 'stddev' and 'variance' in calcs_dict_keys:
+                        continue
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:317]: fix tab

File: dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py [hunk 2]
```diff
@@ -648,6 +649,66 @@ def test_profile(self):
...
+            },
+            quantiles={
+                0: 2.0,
+                1: 3.0,
+                2: 4.0,
+            },
+            num_zeros=0, # default
+            num_negatives=0, # default
+            times=defaultdict(float), # default
+        )
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py:689]: can we delete?

File: dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py [hunk 3]
```diff
@@ -648,6 +649,66 @@ def test_profile(self):
...
+
+        num_profiler.match_count = 0
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py:691]: prob only need to specify hist, sum, and variance. rest is dependent on anything

File: dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py [hunk 4]
```diff
@@ -648,6 +649,66 @@ def test_profile(self):
...
+        num_profiler.histogram_selection = 'auto'
+        num_profiler.histogram_methods['auto']['histogram'] = \
+            mock_report['histogram']
+        num_profiler.quantiles = mock_report['quantiles']
+        num_profiler.times = mock_report['times']
+
+        time_array = [float(i) for i in range(100, 0, -1)]
+        with mock.patch('time.time', side_effect=lambda: time_array.pop()):
+            # Validate that the times dictionary is empty
+            self.assertEqual(defaultdict(float), num_profiler.times)
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py:704]: remove

File: dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py [hunk 5]
```diff
@@ -648,6 +649,66 @@ def test_profile(self):
...
+
+        num_profiler.match_count = 0
+        num_profiler.min = mock_report['min']
+        num_profiler.max = mock_report['max']
+        num_profiler.sum = mock_report['sum']
+        num_profiler.histogram_selection = 'auto'
+        num_profiler.histogram_methods['auto']['histogram'] = \
+            mock_report['histogram']
+        num_profiler.quantiles = mock_report['quantiles']
+        num_profiler.times = mock_report['times']
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py:699]: may not need anything here

File: dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py (hunk 3 above)
Comment (taylorfturner) [reply dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py:691]: yep

File: dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py (hunk 5 above)
Comment (taylorfturner) [reply dataprofiler/tests/profilers/test_numeric_stats_mixin_profile.py:699]: need time and match_count

File: dataprofiler/profilers/numerical_column_stats.py (hunk 1 above)
Comment (taylorfturner) [reply dataprofiler/profilers/numerical_column_stats.py:317]: done

File: dataprofiler/profilers/numerical_column_stats.py [hunk 6]
```diff
@@ -292,6 +292,37 @@ def profile(self):
 
         return profile
 
+    def report(self, remove_disabled_flag):
```
Comment (JGSweets) [inline dataprofiler/profilers/numerical_column_stats.py:295]: should by default be `remove_disabled_flag=False`

Comment (taylorfturner) [reply dataprofiler/profilers/numerical_column_stats.py]: done

**Review by JGSweets — DISMISSED** [review]

File: dataprofiler/profilers/numerical_column_stats.py [hunk 7]
```diff
@@ -292,6 +292,37 @@ def profile(self):
 
         return profile
 
+    def report(self, remove_disabled_flag=False):
+        """
+        Method to call the profile and remove the disabled columns from
+            the profile's report. "Disabled column" is defined as a column
+            that is not prsenet in `self.__calculations` but is present
```
Comment (ksneab7) [inline dataprofiler/profilers/numerical_column_stats.py:299]: Typo in "prsenet"

Comment (taylorfturner) [reply dataprofiler/profilers/numerical_column_stats.py]: fixed

**Review by ksneab7 — APPROVED** [review]

**Review by JGSweets — APPROVED** [review]

**Pull request merged**